### PR TITLE
new easyblock for NCCL (built from source)

### PR DESCRIPTION
**Pull request by branfosj**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyblocks/n/nccl.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        pass
+
+    def build_step(self):
+        """Build NCCL"""
+        # NCCL builds for all supported CUDA compute capabilities by default
+        # If cuda_compute_capabilities or ptx are specified then we override this with the selected options
+
+        # list of CUDA compute capabilities to use can be specifed in two ways (where (2) overrules (1)):
+        # (1) in the easyconfig file, via the custom cuda_compute_capabilities;
+        # (2) in the EasyBuild configuration, via --cuda-compute-capabilities configuration option;
```
Comment (smoors) [inline easybuild/easyblocks/n/nccl.py:57]: to be pedantic, there is a third way, using envar `EASYBUILD_CUDA_COMPUTE_CAPABILITIES`

File: easybuild/easyblocks/n/nccl.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        return extra_vars
+
+    def configure_step(self):
+        """NCCL has no configure step"""
+        pass
+
+    def build_step(self):
+        """Build NCCL"""
+        # NCCL builds for all supported CUDA compute capabilities by default
+        # If cuda_compute_capabilities or ptx are specified then we override this with the selected options
```
Comment (smoors) [inline easybuild/easyblocks/n/nccl.py:53]: @branfosj what's the advantage of having `ptx` as an additional option if you can already override it with `cuda_compute_capabilities`? isn't it confusing to have 2 options that do the same thing?

File: easybuild/easyblocks/n/nccl.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+        # list of CUDA compute capabilities to use can be specifed in two ways (where (2) overrules (1)):
+        # (1) in the easyconfig file, via the custom cuda_compute_capabilities;
+        # (2) in the EasyBuild configuration, via --cuda-compute-capabilities configuration option;
+        cuda_cc = build_option('cuda_compute_capabilities') or self.cfg['cuda_compute_capabilities']
+        if cuda_cc or self.cfg['ptx']:
```
Comment (smoors) [inline easybuild/easyblocks/n/nccl.py:59]: if it's decided that the 2 options (cuda_compute_capabilities and ptx) are needed, we should first remove the duplicates (caps that are in both lists).

**Review by smoors — CHANGES_REQUESTED** [review]

File: easybuild/easyblocks/n/nccl.py (hunk 2 above)
Comment (branfosj) [reply easybuild/easyblocks/n/nccl.py:53]: Based on https://github.com/easybuilders/easybuild-easyblocks/pull/2328#issuecomment-842070642 I've removed the code related to `ptx`.

File: easybuild/easyblocks/n/nccl.py (hunk 3 above)
Comment (branfosj) [reply easybuild/easyblocks/n/nccl.py:59]: Based on https://github.com/easybuilders/easybuild-easyblocks/pull/2328#issuecomment-842070642 I've removed the code related to ptx.

File: easybuild/easyblocks/n/nccl.py (hunk 1 above)
Comment (branfosj) [reply easybuild/easyblocks/n/nccl.py]: done in 2158e90

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @branfosj!

**Pull request merged**